pkg/controller/grafanadashboard/grafana_client.go

```diff
@@ -305,3 +440,49 @@ func newResponse() GrafanaResponse {
 		URL:     &url,
 	}
 }
+
+func (r *GrafanaClientImpl) searchForFolderIDsInDashboards(folderId *int64) ([]GrafanaFolderResponse, error) {
```

Comment:
Needs to be debugged/refactored, expecting to pass in a folder id and search for it in the dashboards, if the response is 0, then assume that no dashboards are assigned to this folder.